projects/go-lib/src/lib/components/go-table/go-table.component.ts

```diff
@@ -8,25 +8,45 @@ import { sortBy } from './go-table-utils';
   templateUrl: './go-table.component.html',
   styleUrls: ['./go-table.component.scss']
 })
-export class GoTableComponent implements OnChanges {
+export class GoTableComponent implements OnInit, OnChanges {
 
   @Input() tableConfig: GoTableConfig;
 
   @ContentChildren(GoTableColumnComponent) columns: QueryList<GoTableColumnComponent>;
 
   localTableConfig: GoTableConfig;
+  showTable: boolean = false;
+
+  ngOnInit() {
+    if (!this.tableConfig) {
+      throw new Error("GoTableComponent: tableConfig is a required Input");
+    } else {
+      this.renderTable();
+    }
+  }
 
   ngOnChanges() {
     this.renderTable();
   }
 
   renderTable() {
-    if (this.tableConfig!.tableData) {
+    if (this.tableConfig) {
       this.localTableConfig = JSON.parse(JSON.stringify(this.tableConfig));
+      this.showTable = true;
       this.handleSort();
+    } else {
+      this.showTable = false;
     }
   }
 
+  hasData() : boolean {
+    if (this.localTableConfig) {
+      return this.localTableConfig.tableData ? this.localTableConfig.tableData.length > 0 : false;
```

Comment:
If `this.localTableConfig.tableData` is anything except 0, it will be true. `0` is falsey, so I _think_ we could write this like so:

```javascript
return this.localTableConfig.tableData.length;
```

Should we make sure `tableData` is present? We could use the TypeScript safe navigator...
```javascript
return this.localTableConfig?.tableData.length
```

Reply:
In this case, I am first checking `this.localTableConfig.tableData` to see if it exists. Then I get the length and evaluate further. This syntax:
`this.localTableConfig?.tableData.length`
 is invalid in TypeScript, there isn't a safe navigator that I'm aware of.

Reply:
I had this wrong... I was referring to the [null assertion operator](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#non-null-assertion-operator) that you mentioned on March 25th in the Dojo.

```
this.localTableConfig!.tableData.length
```

I do see what you're doing, though, and without the null assertion thing, I mean that you don't need the ternary, really, because you can just check `length`. If `length` is any truthy value (0 is falsey), then it would return true, else it would return false. Below would be equivalent functionality, if you wanted to remove the ternary.

```typescript
if (this.localTableConfig && this.localTableConfig.tableData) {
  return this.localeTableConfig.tableData.length;
}

return false;
```

Reply:
All good, I've updated this.